File: Library/ios.py
from Library.finder import *
from plumbum.cmd import xcrun


def delete_derived_data():
    return "rm -rf ~/Library/Developer/Xcode/DerivedData"


simctl = xcrun["simctl"]


def list_simulators():
    return simctl["list"]


def remove_unavailable_simulators():
    return simctl["delete", "unavailable"]


def simulators():
    return list_simulators() | find_between('== Device Types ==', '== Runtimes ==')
    # return f"{list_simulators()} | {find_between('== Device Types ==', '== Runtimes ==')}"


def iphone_simulators():
    return f"{list_simulators()} | {find_between('== Device Types ==', '== Runtimes ==')} | {find_lines_with_word('iPhone')}"

File: src/Core/Library/ios.lua
local finder = require 'library.finder'

local ios = {}

function ios.full_simulators_list()
    return "xcrun simctl list"
end

function ios.ios_simulators()
    return {ios.full_simulators_list(), finder.find_between('== Device Types ==', '== Runtimes =='), finder.find_lines_with_word("iPhone") }
end

return ios